cmd/status/cmdstatus.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package status
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/cli-utils/cmd/status/printers"
+	"sigs.k8s.io/cli-utils/pkg/apply/poller"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling/aggregator"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling/collector"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling/event"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	"sigs.k8s.io/cli-utils/pkg/manifestreader"
+	"sigs.k8s.io/cli-utils/pkg/util/factory"
+)
+
+func GetStatusRunner(f cmdutil.Factory) *StatusRunner {
+	r := &StatusRunner{
+		factory:                    f,
+		inventoryClientFactoryFunc: inventoryClientFactoryFunc,
+		pollerFactoryFunc:          pollerFactoryFunc,
+	}
+	c := &cobra.Command{
+		Use:  "status (DIRECTORY | STDIN)",
+		RunE: r.runE,
+	}
+	c.Flags().DurationVar(&r.period, "poll-period", 2*time.Second,
+		"Polling period for resource statuses.")
+	c.Flags().StringVar(&r.pollUntil, "poll-until", "known",
+		"When to stop polling. Must be one of 'known', 'current', 'deleted', or 'forever'.")
+	c.Flags().StringVar(&r.output, "output", "events", "Output format.")
+	c.Flags().DurationVar(&r.timeout, "timeout", 0,
+		"How long to wait before exiting")
+
+	r.command = c
+	return r
+}
+
+func StatusCommand(f cmdutil.Factory) *cobra.Command {
+	return GetStatusRunner(f).command
+}
+
+// StatusRunner captures the parameters for the command and contains
+// the run function.
+type StatusRunner struct {
+	command *cobra.Command
+	factory cmdutil.Factory
+
+	period    time.Duration
+	pollUntil string
+	timeout   time.Duration
+	output    string
+
+	inventoryClientFactoryFunc func(cmdutil.Factory) (inventory.InventoryClient, error)
```

Comment:
Nit: inventoryClientFactoryFunc is too long. How about invClientFactoryFunc ?

Reply:
done